src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrencePattern.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Collections.Generic;
+
+namespace Microsoft.FeatureManagement.FeatureFilters
+{
+    /// <summary>
+    /// The recurrence pattern specifying how often the time window repeats
+    /// </summary>
+    public class RecurrencePattern
+    {
+        /// <summary>
+        /// The recurrence pattern type
+        /// </summary>
+        public string Type { get; set; }
+
+        /// <summary>
+        /// The number of units between occurrences, where units can be in days, weeks, months, or years, depending on the pattern type
+        /// </summary>
+        public int Interval { get; set; } = 1;
+
+        /// <summary>
+        /// The days of the week on which the time window occurs
+        /// </summary>
+        public IEnumerable<string> DaysOfWeek { get; set; }
+
+        /// <summary>
+        /// The first day of the week.
+        /// </summary>
+        public string FirstDayOfWeek { get; set; } = "Sunday";
```

Comment:
FirstDayOfWeek seems like it should be an enum.

Reply:
In settings enums are modeled as string, but a finite set of options is better represented by an enum in .NET

Reply:
The reason I used string instead of enum here is that
```C#
public object BindParameters(IConfiguration filterParameters)
{
    return filterParameters.Get<TimeWindowFilterSettings>() ?? new TimeWindowFilterSettings();
}
```
If I use enum here, the `BindParameters` will have to parse the string and map it to the enum. In this case, should the validation of the parameters happen in `BindParameters`?


Amer has called out this issue: https://github.com/microsoft/FeatureManagement-Dotnet/pull/266#discussion_r1340432430

Reply:
This should be enum. It's my fault. I underestimated the powerful capabilities of .NET.
At first, I thought that if I used an enum, the filterParameters.Get() wouldn't be able to automatically convert a string in the configuration into the enum. But when I tried it out, I found out that .NET is incredible, it can automatically parse a string into an enum. So, here, we definitly should use enum.